storage/innobase/lock/lock0lock.cc

```diff
@@ -317,6 +316,19 @@ namespace Deadlock
     }
     return nullptr;
   }
+
+  /** Detect deadlock using @ref Deadlock::find_cycle function facility.
+  @param trx Transaction that is waiting for another transaction
+  @return a transaction that is part of a cycle, otherwise nullptr
+  @note the function is not fully idempotent, each call increments a counter */
+  static inline trx_t *detect_deadlock(trx_t *trx)
+  {
+      trx_t *res= nullptr;
+      if ((res = find_cycle(trx)) != nullptr)
+          lock_sys.deadlocks++;
+
+      return res;
```

Comment:
Instead of adding this function and adjusting both callers of `Deadlock::find_cycle()`, it would seem that we can move the `lock_sys.deadlocks++`  to the beginning of `Deadlock::report()`.

Reply:
Fixed